patches/minecraft/net/minecraft/inventory/container/RecipeBookContainer.java.patch

```diff
@@ -1,11 +1,72 @@
 --- a/net/minecraft/inventory/container/RecipeBookContainer.java
 +++ b/net/minecraft/inventory/container/RecipeBookContainer.java
-@@ -31,4 +31,8 @@
+@@ -31,4 +31,69 @@
  
     @OnlyIn(Dist.CLIENT)
     public abstract int func_203721_h();
 +
-+   public java.util.List<net.minecraft.client.util.RecipeBookCategories> getRecipeBookCategories() {
-+      return net.minecraft.client.util.ClientRecipeBook.func_216769_b(this);
++   public net.minecraft.item.crafting.IRecipeType<?> getRecipeType() {
```

Comment:
Just a question:
Why is this done using a bunch of if-switches here?

Somebody from the @Triage-Team: Is it better to make this like he implemented it here. Or should he override the method in the individual containers (like he says modders should do)? Patches would be bigger and have a higher count. However it would be easier to maintain and sets a better example for others.

Reply:
As pointed in this comment in 1.14 implementation https://github.com/MinecraftForge/MinecraftForge/pull/6785#pullrequestreview-434422979 this will allow not to patch enum fields. I will add default statement for throwing exception for notifying when new field will be added

Reply:
It was request from @TheSilkMiner to do it on one place for vanilla containers

Reply:
Since I was mentioned, my line of reasoning was less patches is better. Moreover, it seems Forge is taking the hardcoded approach in other patches and extension interfaces (refer to `IForgeBlock` and the patch for `Block`).

Reply:
I know, but I have seen it in other places the other way around, and since I am starting the reviewing of PRs, i was wondering. I mean we should technically set an example as well for other modders: How todo it right. And this is might or might not be the way we project this kind of solution as right. Hence the request to the Triage team to clarify how this is supposed to be handled.

This was not directed against @yanny7 but more as a question to the Triage team what the going rate here is? Is 4 if-switches okey, 5? 6? or only 3?.

Reply:
Having the logic in one centralised place has its merits - as does having the change spread out over smaller patches to where the change actually makes an effect.
It depends on the code style of the person writing it, and in this case I believe this fits the Forge style the best. If the list were any bigger i'd suggest splitting it out, and that might be worth a //TODO for when (and if) Mojang add any more blocks like these.

Reply:
Perfect, thanks for the clear up.